test/SkipCallsTestHook.t.sol

```diff
@@ -0,0 +1,255 @@
+import {SkipCallsTestHook} from "../src/test/SkipCallsTestHook.sol";
+
+contract SkipCallsTest is Test, Deployers, GasSnapshot {
+    using PoolIdLibrary for PoolKey;
+
+    IPoolManager.SwapParams swapParams =
+        IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+    PoolSwapTest.TestSettings testSettings =
+        PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true, currencyAlreadySent: false});
+
+    function deploy(SkipCallsTestHook skipCallsTestHook) private {
+        SkipCallsTestHook impl = new SkipCallsTestHook();
+        vm.etch(address(skipCallsTestHook), address(impl).code);
+        deployFreshManagerAndRouters();
+        skipCallsTestHook.setManager(IPoolManager(manager));
+
+        (currency0, currency1) = deployMintAndApprove2Currencies();
+
+        assertEq(skipCallsTestHook.counter(), 0);
+
+        (key,) = initPool(currency0, currency1, IHooks(address(skipCallsTestHook)), 3000, SQRT_RATIO_1_1, ZERO_BYTES);
+    }
+
+    function approveAndAddLiquidity(SkipCallsTestHook skipCallsTestHook) private {
+        MockERC20(Currency.unwrap(key.currency0)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        MockERC20(Currency.unwrap(key.currency1)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        modifyLiquidityRouter.modifyLiquidity(key, LIQ_PARAMS, abi.encode(address(this)));
+    }
+
+    function test_beforeInitialize_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 1);
+    }
+
+    function test_afterInitialize_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.BEFORE_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
+                    )
+            )
+        );
+
+        deploy(skipCallsTestHook);
+
+        assertEq(skipCallsTestHook.counter(), 1);
```

Comment:
maybe a comment saying why after `deploy` it would be 1 - i had to go and read that function to know why (same in the test above)